### PR TITLE
chore(telemetry): remove legacy back-compat

### PR DESCRIPTION
**Pull request by bobheadxi**
When V2 telemetry first launched, we had to be careful to support older instances with V1 event logs, and shortly after the launch there were a series of bug fixes that necessitated several compatibility modes from 5.2.0 to 5.2.5, which was released in December 2023. After 5.2.5, we never committed to removing this back-compat as we waited for customers to slowly upgrade past 5.2.5.

Sadly, this came back to haunt us when 5.10.0 came out a few days ago, and our compatibility mode detector [sorted 5.10.0 < 5.2.5](https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1733430008785369?thread_ts=1733425399.918999&cid=C03JR7S7KRP), causing V2T to be exported as old event logs, which prevented them from being exported. After finding the bug and discussing it a bit, we've decided to [remove this back-compat entirely](https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1733507755279819?thread_ts=1733425399.918999&cid=C03JR7S7KRP) since most Cody customers have upgraded past 5.2.5. This simplifies the code significantly.

With the backcompat gone, this basically removes the need for _any_ of the old log-event-to-dotcom stuff, which allowed for the removal for quite a lot of unused code, which simplifies even more stuff.

Closes https://linear.app/sourcegraph/issue/CORE-103/vscodetelemetry-remove-legacy-v1-fallback-exporter

## Test plan

CI, plus a manual test:

Apply a simple patch to make telemetry export in dev without fenangling configuration (this diff is not in this PR):

```diff
diff --git a/vscode/src/services/telemetry-v2.ts b/vscode/src/services/telemetry-v2.ts
index a2857b565..6dcf1e9ad 100644
--- a/vscode/src/services/telemetry-v2.ts
+++ b/vscode/src/services/telemetry-v2.ts
@@ -43,27 +43,11 @@ export function createOrUpdateTelemetryRecorderProvider(
 
             const initialize = telemetryRecorderProvider === undefined
 
-            /**
-             * In testing, send events to the mock server.
-             */
-            if (process.env.CODY_TESTING === 'true') {
-                logDebug(debugLogLabel, 'using mock exporter')
-                updateGlobalTelemetryInstances(
-                    new MockServerTelemetryRecorderProvider({
-                        configuration,
-                        clientState,
-                    })
-                )
-            } else if (isExtensionModeDevOrTest) {
-                logDebug(debugLogLabel, 'using no-op exports')
-                updateGlobalTelemetryInstances(defaultNoOpProvider)
-            } else {
                 updateGlobalTelemetryInstances(
                     new TelemetryRecorderProvider(
                         { configuration, auth, clientState },
                     )
                 )
-            }
 
             /**
              * On first initialization, also record some initial events.
```

Run the extension:

```
pnpm install && pnpm build && cd vscode && pnpm run dev
```

In logs:

```
█ telemetry-v2 recordEvent: cody.auth/connected:
```

in the instance:

image

### CODE REVIEW

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt [hunk 1]
```diff
@@ -74,8 +74,6 @@ interface CodyAgentServer {
   fun featureFlags_getFeatureFlag(params: FeatureFlags_GetFeatureFlagParams): CompletableFuture<Boolean?>
   @JsonRequest("graphql/getCurrentUserCodySubscription")
   fun graphql_getCurrentUserCodySubscription(params: Null?): CompletableFuture<CurrentUserCodySubscription?>
-  @JsonRequest("graphql/logEvent")
-  fun graphql_logEvent(params: Event): CompletableFuture<Null?>
```
Comment (bobheadxi) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/CodyAgentServer.kt:78]: unused by Jetbrains: https://sourcegraph.sourcegraph.com/search?q=r%3A%5Egithub%5C.com%2Fsourcegraph%2Fjetbrains%24+logEvent

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt [hunk 2]
```diff
@@ -7,7 +7,6 @@ object Constants {
   const val Automatic = "Automatic"
   const val Error = "Error"
   const val Finished = "Finished"
-  const val IDEEXTENSION = "IDEEXTENSION"
```
Comment (bobheadxi) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/Constants.kt:10]: unused by Jetbrains: https://sourcegraph.sourcegraph.com/search?q=r:%5Egithub%5C.com/sourcegraph/jetbrains%24+IDEEXTENSION&patternType=keyword&sm=0

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ExtensionConfiguration.kt [hunk 3]
```diff
@@ -13,7 +13,6 @@ data class ExtensionConfiguration(
   val verboseDebug: Boolean? = null,
   val telemetryClientName: String? = null,
   val codebase: String? = null,
-  val eventProperties: EventProperties? = null,
```
Comment (bobheadxi) [inline agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ExtensionConfiguration.kt:16]: Unused by Jetbrains: https://sourcegraph.sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/jetbrains%24+EventProperties&patternType=keyword&sm=0

**Review by abeatrix — APPROVED** [review]
ty!

**Review by dadlerj — APPROVED** [review]

**Comment by bobheadxi** [comment]
We are go for merge: https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1733507755279819?thread_ts=1733425399.918999&cid=C03JR7S7KRP

**Comment by bobheadxi** [comment]
Slack discussion about how to merge this: https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1733508306643029

**Comment by bobheadxi** [comment]
backports are up:

* https://github.com/sourcegraph/cody/pull/6276
* https://github.com/sourcegraph/cody/pull/6275
* https://github.com/sourcegraph/cody/pull/6274

**Pull request merged**